### DISCUSSION TITLE
how can I use mini.move in mini.nvim plugin to replace shift left << with one <?

### DISCUSSION THREAD

**Discussion by Lslightly** [LazyVim/LazyVim#2250] [Q&A] [answered]
additional information:
I'm not familiar with lazyvim's plugin layout and thus I don't know where to add mapping and options as shown in https://github.com/echasnovski/mini.nvim/blob/main/readmes/mini-move.md#default-config. I also don't know where to add `require('mini.move').setup()` shown after *Important* in https://github.com/echasnovski/mini.nvim/blob/main/readmes/mini-move.md#installation

**Comment by dpetka2001** [comment]
You should really read the [docs](https://www.lazyvim.org/configuration#-file-structure) to understand how you need to structure your files.
You could do the following
```lua
-- ~/.config/nvim/lua/plugins/move.lua
return {
  {
    "echasnovski/mini.move",
    event = "VeryLazy",
    opts = {},
  },
}
```
You could map to move the lines with `<`, but I would suggest against it, since it's an internal mapping of Neovim to 
indent and if you decide to use it to shift lines there will be a small, but observable, delay for the action to complete. 
Better use `mini.move`'s default mappings.

**Reply by Lslightly** [reply]
Thank you. And I'm sorry for my poor documentation reading and question description. If I understand correctly, `The files under config will be automatically loaded at the appropriate time, so you don't need to require those files manually.` in [configuration documentation](https://www.lazyvim.org/configuration#-file-structure) says there's no need to add `require(...).setup()`, right? Besides, where can I place the `mapping` as shown in [mini.move README](https://github.com/echasnovski/mini.move/blob/main/README.md)? For your convenience, I put the default config as follows.

```
-- No need to copy this inside `setup()`. Will be used automatically.
{
  -- Module mappings. Use `''` (empty string) to disable one.
  mappings = {
    -- Move visual selection in Visual mode. Defaults are Alt (Meta) + hjkl.
    left = '<M-h>',
    right = '<M-l>',
    down = '<M-j>',
    up = '<M-k>',

    -- Move current line in Normal mode
    line_left = '<M-h>',
    line_right = '<M-l>',
    line_down = '<M-j>',
    line_up = '<M-k>',
  },

  -- Options which control moving behavior
  options = {
    -- Automatically reindent selection during linewise vertical move
    reindent_linewise = true,
  },
}
```

Is it correct to write the above settings in 'opts' in `move.lua`? Or somewhere else?

**Reply by dpetka2001** [reply, marked as answer]
You should put the `mappings` inside `opts`. You can still use the method with `require(...).setup()`, but you should do that in the `config` function and not `opts` if you'd like to do that. So, something like
```lua
return {
  {
    "echasnovski/mini.move",
    event = "VeryLazy",
    opts = {
      mappings = {
        left = '<',
      },
    },
  },
}
```
is equivalent with
```lua
return {
  {
    "echasnovski/mini.move",
    event = "VeryLazy",
    config = function()
      require(...).setup({
        mappings = {
          left = '<',
        },
      })
    end,
  }
}
```

**Reply by Lslightly** [reply]
Replacing `<M-h>` with '<' is more in line with the discussion title. Would you please edit the reply above?
